ndc-models/src/lib.rs

```diff
@@ -420,9 +420,37 @@ pub struct Query {
     pub offset: Option<u32>,
     pub order_by: Option<OrderBy>,
     pub predicate: Option<Expression>,
+    ///
+    pub groups: Option<Grouping>,
 }
 // ANCHOR_END: Query
 
+// ANCHOR: Grouping
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Grouping")]
+pub struct Grouping {
+    /// Dimensions along which to partition the data
+    pub dimensions: IndexMap<String, Dimension>,
+    /// Fields to fetch in each group
+    pub fields: IndexMap<String, Field>,
+    /// Aggregates to compute in each group
+    pub aggregates: IndexMap<String, Aggregate>,
+}
+// ANCHOR_END: Grouping
+
+// ANCHOR: Dimension
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[skip_serializing_none]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Dimension")]
+pub enum Dimension {
+    Column {
+        column_name: String,
```

Comment:
We'd need a column path thing here to address grouping over nested object fields.